assets/src/dashboard/components/toaster/provider.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  createContext,
+  useMemo,
+  useCallback,
+  useState,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+
+export const ToasterContext = createContext(null);
+
+const ToastProvider = ({ children }) => {
+  const [toasts, setToasts] = useState({});
+
+  const activeToasts = useMemo(
+    () => Object.values(toasts).filter((toast) => toast.isActive),
+    [toasts]
+  );
+
+  const resetToasts = useCallback(() => {
+    setToasts({});
+  }, [setToasts]);
+
+  const removeToast = useCallback(
+    (id) =>
+      id &&
+      setToasts((existingToasts) => ({
+        ...existingToasts,
+        [id]: { ...existingToasts[id], isActive: false },
+      })),
+    []
+  );
+
+  const addToast = useCallback(
+    ({ message, severity, id }) =>
+      setToasts((existingToasts) => ({
+        ...existingToasts,
+        [id]: { message, severity, id, isActive: true },
+      })),
+    []
+  );
+
+  useEffect(() => resetToasts, [resetToasts]);
```

Comment:
Is this effect needed?  It seems like it will only ever run on the initial render since the reference to `resetToasts` should never change and also the initial state of `toasts` is already `{}`

Reply:
nope! i don't think we need this anymore with the cleaned up toaster. thanks for the eyes. I removed it.

Reply:
I thought the point of this was just to reset the toasts on unmount for the provider, which makes sense for only running on initial render. 

But now thinking about it, I'm wondering wondering why we need to reset the toasts when this unmounts.